web/src/engine/websites/LeerCapitulo.ts

```diff
@@ -0,0 +1,57 @@
+import { Tags } from '../Tags';
+import icon from './LeerCapitulo.webp';
+import { DecoratableMangaScraper, type Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchRequest, FetchWindowScript } from '../FetchProvider';
+
+const pageScript = `
+    new Promise(resolve => {
+        const data = document.querySelector('p#array_data').textContent.trim();
+        resolve(atob(data.replace(
+          /[A-Z0-9]/gi,
+          (_0xe528ce) =>
+            '13RjUMgYBObXQDatoWz8TIsmZN7Pq6vSFywnLA04iC9kdupEhfGxVec5Kl2JrH'[
+              'p1iXCxTFYQKEMyG0U5mwW29VsRSAecuok6zgnJNOq7bvtfBZPa3rHjDlIhL48d'.indexOf(
+                _0xe528ce
+              )
+            ]
+        )).split(','));
+    });
+`;
+
+@Common.MangaCSS(/^{origin}\/manga\/[^/]+\/[^/]+\/$/, 'div.media-body .title-manga')
+@Common.ChaptersSinglePageCSS('div.chapter-list ul li h4 a')
+@Common.PagesSinglePageJS(pageScript)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+
+    private readonly categories = ['/completed/', '/ongoing/', '/paused/', '/cancelled/'];
+
+    public constructor() {
+        super('leercapitulo', 'LeerCapitulo', 'https://www.leercapitulo.com', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Spanish, Tags.Source.Aggregator);
+    }
+
+    public override async Initialize(): Promise<void> {
+        const uri = new URL(this.URI);
+        const request = new FetchRequest(uri.href);
+        await FetchWindowScript(request, `localStorage.setItem('display_mode', '1')`, 1500);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        let mangaList = [];
+        for (const cat of this.categories) {
```

Comment:
Avoid misleading shortening of variable names, no need to save memory => `category` instead of 🐈